### PR TITLE
Caching redo

### PR DESCRIPTION
**Pull request by lbjay**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -234,6 +234,18 @@ As of v0.4.0 you can toggle the maintenance mode on a host.
             host.set_maintenance(True)
 
 
+**Caching**
+
+As of v0.7.0 the use of requests_cache has been dropped in favor of an internal
+cache that stores the responses from a subset of the single-item endpoint
+methods. A future goal is to allow more granular control over the caching policy,
+but for now the following endpoint responses are cached:
```
Comment (djcp) [inline README.rst]: Can you describe the caching rules you've implemented? How do you clear the cache? Are there easy ways to disable caching all together for development / debugging?

Comment (djcp) [inline README.rst]: This is a per-process memory cache?

Comment (lbjay) [inline README.rst]: Added a few additional notes to the README

Comment (lbjay) [inline README.rst]: Yes, per-process, in-memory cache. In an ideal world a user could provide their own caching by passing in something that follows the basic get/set api, but I can't get motivated for that while I'm pretty much still the only user (AFAIK).

**Pull request merged**